c7n/resources/rds.py

```diff
@@ -281,6 +301,68 @@ def process(self, dbs):
                 **params)
 
 
+@filters.register('upgrade-available')
+class UpgradeAvailable(Filter):
+    """ Scan DB instances for available engine upgrades
+
+    This will pull DB instances & check their specific engine for any
+    engine version with higher release numbers than the current one
+
+    This will also tag the rds instance with 'target_engine' which is
```

Comment:
s/tag/annotate.. tag has a very specific meaning in aws, which is not what's being done here.